src/Controls/src/Core/Handlers/Items2/iOS/CarouselViewController2.cs

```diff
@@ -254,14 +254,28 @@ void CollectionViewUpdating(object sender, NotifyCollectionChangedEventArgs e)
 		[UnconditionalSuppressMessage("Memory", "MEM0003", Justification = "Proven safe in test: MemoryTests.HandlerDoesNotLeak")]
 		void CollectionViewUpdated(object sender, NotifyCollectionChangedEventArgs e)
 		{
+			int targetPosition;
 			if (_positionAfterUpdate == -1)
 			{
 				return;
 			}
 
 			//_gotoPosition = -1;
 
-			var targetPosition = _positionAfterUpdate;
+			// We need to update the position while modifying the collection.
+			if (ItemsView.ItemsUpdatingScrollMode == ItemsUpdatingScrollMode.KeepItemsInView)
+			{
+				targetPosition = 0;
```

Comment:
won t this always scroll to the first position ?

Reply:
> won t this always scroll to the first position ?

Yes, based on the [documentation](https://learn.microsoft.com/en-us/dotnet/maui/user-interface/controls/carouselview/scrolling?view=net-maui-9.0#control-scroll-position-when-new-items-are-added) and behavior of the Windows platform, we implemented code changes to update the `scrollposition` according to the `ItemsUpdatingScrollMode`. When `ItemsUpdatingScrollMode` is set to `KeepItemsInView`, the scroll position will adjust to the first item in view. To maintain the scrolled index, `KeepScrollOffset` should be used instead.